c++/src/linked-list.cc

```diff
@@ -174,31 +174,23 @@ class LinkedList {
     }
 
     Node<T>* lookingAt = this->head;
-    Node<T>* nodeToDelete = nullptr;
```

Comment:
Ah okay I think I see. The condition you used in the `while` is fine too because it conveys much more than `while (true)`, and it apparently has the added benefit of removing the need for this variable, though that in and of itself should not be a motivating reason alone.

What if `this->head` is `nullptr`? Then the initial check of the while condition `lookingAt->getValue()` will be attempting to dereference a null pointer.

If you want to keep it this way, you can change it to be something like:

``` cpp
while (lookingAt != nullptr && lookingAt->getValue() != value)
```